internal/compiler/fileloader.go

```diff
@@ -225,15 +225,11 @@ func (t *parseTask) start(loader *fileLoader) {
 }
 
 func (p *fileLoader) loadSourceFileMetaData(path tspath.Path) {
-	p.sourceFileMetaDatasMutex.Lock()
+	p.sourceFileMetaDatasMutex.RLock()
 	if _, ok := p.sourceFileMetaDatas[path]; ok {
 		return
```

Comment:
```suggestion
		p.sourceFileMetaDatasMutex.RUnlock()
		return
```

This is a deadlock otherwise.

Reply:
Gosh I'm really messing up this mutex, I've gotten unproductively excited with the landing close